fetch-in-react-practice/client/components/app.jsx

```diff
@@ -116,8 +116,8 @@ export default class App extends React.Component {
         return response.json();
       })
       .then(data => {
-        const todosCopy = this.state.todos;
-        todosCopy[toggledTodoIndex] = data;
+        const todosCopy = [...this.state.todos];
+        todosCopy[toggledTodoIndex].isCompleted = todo.isCompleted;
```

Comment:
This is actually modifying the original object in the array. Instead, replace it like you were already doing

```js
todosCopy[toggledIndex] = data;
```

The idea is to always be swapping objects completely. If that object is in an array, you will swap the array too, just leave the unaffected stuff in the copy.

Reply:
okay i think i got it right this time.